src/pages/home/sidebar/SidebarScreen.js

```diff
@@ -1,56 +1,192 @@
-import React from 'react';
-import {View} from 'react-native';
-import PropTypes from 'prop-types';
-import styles from '../../../styles/styles';
+import React, {Component} from 'react';
+import {View, Animated} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import styles, {getNavigationMenuStyle} from '../../../styles/styles';
 import SidebarLinks from './SidebarLinks';
 import CreateMenu from '../../../components/CreateMenu';
 import FAB from '../../../components/FAB';
 import ScreenWrapper from '../../../components/ScreenWrapper';
+import {
+    hide as hideSidebar,
+    show as showSidebar,
+} from '../../../libs/actions/Sidebar';
+import variables from '../../../styles/variables';
+import {redirect} from '../../../libs/actions/App';
+import ROUTES from '../../../ROUTES';
+import Timing from '../../../libs/actions/Timing';
+import withWindowDimensions, {windowDimensionsPropTypes} from '../../../components/withWindowDimensions';
+import CONST from '../../../CONST';
+import compose from '../../../libs/compose';
+import ONYXKEYS from '../../../ONYXKEYS';
 
 const propTypes = {
-    // Toggles the navigation menu open and closed
-    onLinkClick: PropTypes.func.isRequired,
+    // propTypes for withWindowDimensions
+    ...windowDimensionsPropTypes,
+};
 
-    // Current state of the CreateMenu component (active or inactive)
-    isCreateMenuActive: PropTypes.bool.isRequired,
+class SidebarScreen extends Component {
+    constructor(props) {
+        super(props);
 
-    // Callback to fire on request to toggle the CreateMenu
-    toggleCreateMenu: PropTypes.func.isRequired,
+        this.onCreateMenuItemSelected = this.onCreateMenuItemSelected.bind(this);
+        this.toggleCreateMenu = this.toggleCreateMenu.bind(this);
+        this.showNavigationMenu = this.showNavigationMenu.bind(this);
+        this.recordTimerAndHideSidebar = this.recordTimerAndHideSidebar.bind(this);
+        this.navigateToSettings = this.navigateToSettings.bind(this);
 
-    // Callback to fire when a CreateMenu item is selected
-    onCreateMenuItemSelected: PropTypes.func.isRequired,
+        this.state = {
+            isCreateMenuActive: false,
+        };
 
-    // Callback to fire on avatar click
-    onAvatarClick: PropTypes.func.isRequired,
-};
+        this.animation = new Animated.Value(
```

Comment:
Confused on what this does, does this just start a navigation only if the sidebar is shown?

Reply:
The `Animated.Value` are kind of magic. But if you see one it most likely means that an animated component is using this value to animate some style property over time. In this case, we are using this value to set the `translateX` initial state. But this by itself doesn't start any animations. For that we use `Animated.Timing` and then pass this value into that method. More info [here](https://reactnative.dev/docs/animated).